include/vcpkg/base/stringview.h

```diff
@@ -39,23 +33,56 @@ namespace vcpkg
         constexpr size_t size() const noexcept { return m_size; }
         constexpr bool empty() const noexcept { return m_size == 0; }
 
+        // intentionally not provided because this may not be null terminated
+        // constexpr const char* c_str() const
+
         std::string to_string() const;
         void to_string(std::string& out) const;
         explicit operator std::string() const { return to_string(); }
 
         StringView substr(size_t pos, size_t count = std::numeric_limits<size_t>::max()) const noexcept;
 
         constexpr char operator[](size_t pos) const noexcept { return m_ptr[pos]; }
```

Comment:
Should this return a `const char&`?

Reply:
When it's cheap to return a prvalue I think we should return the prvalue since it avoids aliasing mess but I don't feel strongly about this.